src/Entities/Entity.cpp

```diff
@@ -1546,6 +1546,9 @@ bool cEntity::DoMoveToWorld(cWorld * a_World, bool a_ShouldSendRespawn, Vector3d
 	// Take note of old chunk coords
 	auto OldChunkCoords = cChunkDef::BlockToChunk(GetPosition());
 
+	// Prevent the entity from teleporting back immediately, in case the destination is a portal
+	m_PortalCooldownData.m_ShouldPreventTeleportation = true;
```

Comment:
This should not be needed, it should already be set when the entity is detected inside of the portal.

Reply:
Ah, now that you mention it I'd added those while trying to figure out why the player was spawning on top of the portals and then didn't realize it could be safely removed now (which I just tested and it works fine).